lib/prompb/prompb.go

```diff
@@ -60,32 +50,85 @@ type Label struct {
 	Value string
 }
 
-// UnmarshalProtobuf unmarshals wr from src.
+// LabelsToString converts labels to Prometheus-compatible string
+func LabelsToString(labels []Label) string {
+	labelsCopy := append([]Label{}, labels...)
+	sort.Slice(labelsCopy, func(i, j int) bool {
+		return string(labelsCopy[i].Name) < string(labelsCopy[j].Name)
+	})
+	var b []byte
+	b = append(b, '{')
+	for i, label := range labelsCopy {
+		if len(label.Name) == 0 {
+			b = append(b, "__name__"...)
+		} else {
+			b = append(b, label.Name...)
+		}
+		b = append(b, '=')
+		b = strconv.AppendQuote(b, label.Value)
+		if i < len(labels)-1 {
+			b = append(b, ',')
+		}
+	}
+	b = append(b, '}')
+	return string(b)
+}
+
+type WriteRequestUnmarshaller struct {
+	wr *WriteRequest
```

Comment:
It is better to store wr field by value instead of by pointer. This will remove the need for a memory allocation and a check for nil wr in the code below. This also removes and additional pointer dereferencing when accessing wr.

Reply:
fixed, thx